protocols/rendezvous/src/server.rs

```diff
@@ -168,44 +167,63 @@ impl NetworkBehaviour for Behaviour {
         }
 
         let poll_res = self.inner.poll(cx, params);
-        if let Poll::Ready(ToSwarm::GenerateEvent(event)) = poll_res {
-            let opt_event = match event {
-                libp2p_request_response::Event::Message {
-                    peer: peer_id,
-                    message:
-                        libp2p_request_response::Message::Request {
-                            request, channel, ..
-                        },
-                } => {
-                    let (event, response) =
-                        handle_request(peer_id, request, &mut self.registrations);
-                    if let Some(resp) = response {
-                        self.inner
-                            .send_response(channel, resp)
-                            .expect("Send response");
-                    }
+        if let Poll::Ready(to_swarm) = poll_res {
+            match to_swarm {
+                ToSwarm::GenerateEvent(event) => {
+                    let opt_event = match event {
+                        libp2p_request_response::Event::Message {
+                            peer: peer_id,
+                            message:
+                                libp2p_request_response::Message::Request {
+                                    request, channel, ..
+                                },
+                        } => {
+                            let (event, response) =
+                                handle_request(peer_id, request, &mut self.registrations);
+                            if let Some(resp) = response {
+                                self.inner
+                                    .send_response(channel, resp)
+                                    .expect("Send response");
+                            }
+
+                            Some(event)
+                        }
+                        libp2p_request_response::Event::ResponseSent { .. } => None,
+                        libp2p_request_response::Event::InboundFailure {
+                            peer,
+                            request_id,
+                            error,
+                        } => {
+                            log::warn!(
+                                "Inbound request {request_id} with peer {peer} failed: {error}"
+                            );
+
+                            None
```

Comment:
Same as for the client, it is important that we `loop` here, otherwise we are breaking the poll-fn contract[^1].

[^1]: Must only return `Poll::Pending` if an inner component returned `Poll::Pending` or we registered a waker.